flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java

```diff
@@ -763,6 +763,68 @@ private static List<TestSetSpec> allTypesBasic() {
                         // ROW
                         // RAW
                         .build(),
+                CastTestSpecBuilder.testCastTo(TIME(2))
+                        .fromCase(TIME(), null, null)
+                        .failRuntime(CHAR(3), "foo", DateTimeException.class)
+                        .failRuntime(VARCHAR(5), "Flink", DateTimeException.class)
+                        .failRuntime(STRING(), "Flink", DateTimeException.class)
+                        .fromCase(STRING(), "123", LocalTime.of(23, 0, 0))
+                        .fromCase(STRING(), "123:45", LocalTime.of(23, 45, 0))
+                        .failRuntime(STRING(), "2021-09-27", DateTimeException.class)
+                        .failRuntime(STRING(), "2021-09-27 12:34:56", DateTimeException.class)
```

Comment:
I am wondering if we should duplicate those. Internally, there is no difference if we cast those cases to `TIME(2)` or `TIME(0)`.

Personally, I'd be leaning to leave only the cases where we truncate the precision.